### PR TITLE
Remove set methods from socket factory

### PR DESCRIPTION
**Pull request by sazzad16**

### CODE REVIEW

File: src/main/java/redis/clients/jedis/JedisSocketFactory.java [hunk 1]
```diff
@@ -29,24 +29,12 @@
   @Deprecated
   String getHost();
 
-  @Deprecated
-  void setHost(String host);
-
   @Deprecated
```
Comment (gkorland) [inline src/main/java/redis/clients/jedis/JedisSocketFactory.java:32]: @sazzad16 why did you remove only some of the deprecated methods?

Comment (sazzad16) [reply src/main/java/redis/clients/jedis/JedisSocketFactory.java:32]: @gkorland I'm 💯 that we don't need `set` methods at least in this interface. (Reference: Connection.java)
But in case of get methods, there's little left to be 💯. (Reference: DefaultJedisSocketFactory.java)

Comment (sazzad16) [reply src/main/java/redis/clients/jedis/JedisSocketFactory.java:32]: @gkorland I'm 100% convinced that we don't need set methods at least in this interface.
But in case of get methods I'm about 90% convinced. 10% uncertainty is because of codes in DefaultJedisSocketFactory.createSocket().

**Comment by sazzad16** [comment]
Resolved by https://github.com/redis/jedis/commit/858c8051301854e8343eef66aa68c6b9dab1c328
